### PR TITLE
Add Cloud Based Hybrid Approach

### PR DESCRIPTION
**Pull request by dinuish94**

### CODE REVIEW

File: plugins/org.wso2.developerstudio.eclipse.esb.cloud/plugin.xml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+-->
+
+<?eclipse version="3.4"?>
+<plugin>
+    <extension point="org.eclipse.ui.popupMenus">
+        <objectContribution objectClass="org.eclipse.core.resources.IProject" 
+        id="org.wso2.developerstudio.eclipse.esb.cloud.deploytocloud">
+            <action
+                label="Deploy to Integration Cloud"
+                icon="icons/car.png"
```
Comment (dakshika) [inline plugins/org.wso2.developerstudio.eclipse.esb.cloud/plugin.xml]: need to change the icon

File: plugins/org.wso2.developerstudio.eclipse.esb.cloud/resources/init.sh [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+# limitations under the License
+# ------------------------------------------------------------------------
+set -e
+
+# volume mounts
+config_volume=${WORKING_DIRECTORY}/wso2-config-volume
+artifact_volume=${WORKING_DIRECTORY}/wso2-artifact-volume
+
+# a grace period for mounts to be setup
+echo "Waiting for all volumes to be mounted..."
```
Comment (dakshika) [inline plugins/org.wso2.developerstudio.eclipse.esb.cloud/resources/init.sh]: Seems this is not relevant to this PR

File: plugins/org.wso2.developerstudio.eclipse.esb.cloud/src/org/wso2/developerstudio/eclipse/esb/cloud/client/IntegrationCloudServiceClient.java [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+public class IntegrationCloudServiceClient {
+    
+    private static CookieStore cookieStore;
+    private static IntegrationCloudServiceClient client;
+    
+    private static final String NO_RESOURCES_ERROR = "Application creation request not accepted because no resources available to create new application/version";
+    private static final String VERSION_EXISTS_ERROR = "applicationRevision is required!";
```
Comment (dakshika) [inline plugins/org.wso2.developerstudio.eclipse.esb.cloud/src/org/wso2/developerstudio/eclipse/esb/cloud/client/IntegrationCloudServiceClient.java]: Make this human readable

File: plugins/org.wso2.developerstudio.eclipse.esb.cloud/src/org/wso2/developerstudio/eclipse/esb/cloud/client/IntegrationCloudServiceClient.java [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+        data.put("password", password);
+
+        String response = HTTPClientUtil.sendPostWithFormData(loginUrl, new HashMap<String, String>(), data, cookieStore);
+        
+        JsonParser parser = new JsonParser();
+        JsonElement jsonTree = parser.parse(response);
+        
+        String message = jsonTree.getAsJsonObject().get("message").getAsString();
+        
+        return message.equals("User successfully logged in");
```
Comment (dakshika) [inline plugins/org.wso2.developerstudio.eclipse.esb.cloud/src/org/wso2/developerstudio/eclipse/esb/cloud/client/IntegrationCloudServiceClient.java]: You have been successfully  logged in

File: plugins/org.wso2.developerstudio.eclipse.esb.cloud/src/org/wso2/developerstudio/eclipse/esb/cloud/exceptions/HttpClientException.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
+package org.wso2.developerstudio.eclipse.esb.cloud.exceptions;
```
Comment (rosensilva) [inline plugins/org.wso2.developerstudio.eclipse.esb.cloud/src/org/wso2/developerstudio/eclipse/esb/cloud/exceptions/HttpClientException.java]: Shall we have a license header here?

File: plugins/org.wso2.developerstudio.eclipse.esb.cloud/src/org/wso2/developerstudio/eclipse/esb/cloud/client/IntegrationCloudServiceClient.java [hunk 6]
```diff
@@ -0,0 +1,288 @@
...
+        data.put("userName", user);
+        data.put("password", password);
+
+        String response = HTTPClientUtil.sendPostWithFormData(loginUrl, new HashMap<String, String>(), data,
+                cookieStore);
+
+        JsonParser parser = new JsonParser();
+        JsonElement jsonResponse = parser.parse(response);
+
+        String message = jsonResponse.getAsJsonObject().get("message").getAsString();
```
Comment (chanikag) [inline plugins/org.wso2.developerstudio.eclipse.esb.cloud/src/org/wso2/developerstudio/eclipse/esb/cloud/client/IntegrationCloudServiceClient.java]: Can jsonResponse.getAsJsonObject().get("message") null?

File: plugins/org.wso2.developerstudio.eclipse.esb.cloud/src/org/wso2/developerstudio/eclipse/esb/cloud/client/IntegrationCloudServiceClient.java [hunk 7]
```diff
@@ -0,0 +1,288 @@
...
+        return message.equals(CloudServiceConstants.ResponseMessages.USER_SUCCESSFULLY_LOGGED_IN);
```
Comment (chanikag) [inline plugins/org.wso2.developerstudio.eclipse.esb.cloud/src/org/wso2/developerstudio/eclipse/esb/cloud/client/IntegrationCloudServiceClient.java]: Shall we switch the parameters for equals method to make sure it doesn't throw NPE?

File: plugins/org.wso2.developerstudio.eclipse.esb.cloud/src/org/wso2/developerstudio/eclipse/esb/cloud/client/IntegrationCloudServiceClient.java (hunk 6 above)
Comment (dinuish94) [reply plugins/org.wso2.developerstudio.eclipse.esb.cloud/src/org/wso2/developerstudio/eclipse/esb/cloud/client/IntegrationCloudServiceClient.java]: Fixed in https://github.com/wso2/devstudio-tooling-esb/pull/1063

File: plugins/org.wso2.developerstudio.eclipse.esb.cloud/src/org/wso2/developerstudio/eclipse/esb/cloud/client/IntegrationCloudServiceClient.java (hunk 7 above)
Comment (dinuish94) [reply plugins/org.wso2.developerstudio.eclipse.esb.cloud/src/org/wso2/developerstudio/eclipse/esb/cloud/client/IntegrationCloudServiceClient.java]: Fixed in https://github.com/wso2/devstudio-tooling-esb/pull/1063

**Pull request merged**